### PR TITLE
Test: Move some blueprints 'down', closer to test_document_service

### PR DESCRIPTION
**Pull request by pbullhove**
## What does this pull request change?

Moved blueprints down into document service. 
Made the example more 'concrete' by using everyday names such as cat, owner, person, etc. 
- Also had to add some more stuff to the mock recipe provider due to this.

### CODE REVIEW

File: src/tests/unit/services/document_service/test_remove.py [hunk 1]
```diff
@@ -40,98 +40,110 @@ def setUp(self):
...
         self.storage = {
             "1": {
                 "_id": "1",
-                "name": "Parent",
-                "type": "all_contained_cases_blueprint",
-                "nested": {
-                    "name": "Nested",
-                    "type": "all_contained_cases_blueprint",
-                    "nested": {
+                "name": "",
```
Comment (collinlokken) [inline src/tests/unit/services/document_service/test_remove.py:141]: Maybe have a name for clarity?

File: src/tests/unit/services/document_service/test_remove.py [hunk 2]
```diff
@@ -153,14 +165,14 @@ def test_remove_document_also_removes_storage_uncontained_grand_children(self):
...
-                "type": "all_contained_cases_blueprint",
-                "nested": {
-                    "name": "Nested",
-                    "type": "all_contained_cases_blueprint",
-                    "nested": {
-                        "name": "Parent",
-                        "type": "all_contained_cases_blueprint",
+                "name": "",
+                "type": "Person",
+                "...": {
```
Comment (collinlokken) [inline src/tests/unit/services/document_service/test_remove.py:170]: to me this shows as "...", is this intended?

**Review by collinlokken — CHANGES_REQUESTED** [review]
Nice to finally get a bit more understandable tests and names 🤩 I would remove the "...", and maybe add value to the `name` attribute where name is an empty string. Finally it would be nice to rename the `containement_example` folder to something more generic.

**Review by collinlokken — CHANGES_REQUESTED** [review]

File: src/tests/unit/common/utils/validators/test_validate_entity.py [hunk 3]
```diff
@@ -274,7 +275,13 @@ def test_array_with_complex_child_getting_array_type_from_parent_node(self):
             "customers": [],
         }
 
-        parent_node = tree_node_from_dict(test_entity, self.mock_document_service.get_blueprint)
+        parent_node = tree_node_from_dict(
+            test_entity,
+            self.mock_document_service.get_blueprint,
+            recipe_provider=MockStorageRecipeProvider(
```
[the remark this answers is not in the document]
Comment (pbullhove) [reply src/tests/unit/common/utils/validators/test_validate_entity.py]: done

File: src/tests/unit/services/document_service/test_remove.py [hunk 4]
```diff
@@ -153,14 +165,14 @@ def test_remove_document_also_removes_storage_uncontained_grand_children(self):
...
-                    "type": "all_contained_cases_blueprint",
-                    "nested": {
-                        "name": "Parent",
-                        "type": "all_contained_cases_blueprint",
+                "name": "",
+                "type": "Person",
+                "not_in_blueprint": {
+                    "name": "John",
+                    "type": "Person",
+                    "not_in_blueprint": {
```
[the remark this answers is not in the document]
Comment (pbullhove) [reply src/tests/unit/services/document_service/test_remove.py:173]: done

**Review by collinlokken — APPROVED** [review]
Very nice!

**Pull request merged**